### DISCUSSION TITLE
Fine-tuning PaddleOCR Detection Model - Can I use default PP-OCRv5_mobile_det.yml config for custom dataset?

### DISCUSSION THREAD

**Discussion by RunSongham** [PaddlePaddle/PaddleOCR#16614] [Q&A]
Question:
I'm planning to fine-tune the PaddleOCR detection model on a custom dataset and have a question about the configuration settings.
My Dataset Specifications:

Image size: 520 × 110 pixels
![image_dx_0](https://github.com/user-attachments/assets/3d3f1a11-c2c2-418c-9bff-253732d5b416)

Label format:
images/image_dx_0.jpg    [{"transcription": "부", "points": [[25, 13], [85, 13], [85, 55], [25, 55]], "difficult": false}, {"transcription": "산", "points": [[25, 55], [85, 55], [85, 97], [25, 97]], "difficult": false}, {"transcription": "99아1571", "points": [[85, 13], [497, 13], [497, 96], [85, 96]], "difficult": false}]

My Question:
Can I use the default settings in PP-OCRv5_mobile_det.yml as-is for training the detection model, or do I need to modify certain parameters (such as input size, anchor sizes, etc.) to accommodate my custom dataset's dimensions (520 × 110)?
If modifications are necessary, which specific parameters should I adjust for optimal performance with this image size and text layout?
Any guidance would be greatly appreciated!

**[PP-OCRv5_mobile_det.yml]**
Global:
  model_name: PP-OCRv5_mobile_det # To use static model for inference.
  debug: false
  use_gpu: true
  epoch_num: &epoch_num 500
  log_smooth_window: 20
  print_batch_step: 100
  save_model_dir: ./output/PP-OCRv5_mobile_det
  save_epoch_step: 10
  eval_batch_step:
  - 0
  - 1500
  cal_metric_during_train: false
  checkpoints:
  pretrained_model: https://paddleocr.bj.bcebos.com/pretrained/PPLCNetV3_x0_75_ocr_det.pdparams
  save_inference_dir: null
  use_visualdl: false
  infer_img: doc/imgs_en/img_10.jpg
  save_res_path: ./checkpoints/det_db/predicts_db.txt
  d2s_train_image_shape: [3, 640, 640]
  distributed: true

Architecture:
  model_type: det
  algorithm: DB
  Transform: null
  Backbone:
    name: PPLCNetV3
    scale: 0.75
    det: True
  Neck:
    name: RSEFPN
    out_channels: 96
    shortcut: True
  Head:
    name: DBHead
    k: 50
    fix_nan: True

Loss:
  name: DBLoss
  balance_loss: true
  main_loss_type: DiceLoss
  alpha: 5
  beta: 10
  ohem_ratio: 3

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001 #(8*8c)
    warmup_epoch: 2
  regularizer:
    name: L2
    factor: 5.0e-05

PostProcess:
  name: DBPostProcess
  thresh: 0.3
  box_thresh: 0.6
  max_candidates: 1000
  unclip_ratio: 1.5

Metric:
  name: DetMetric
  main_indicator: hmean

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/icdar2015/text_localization/
    label_file_list:
      - ./train_data/icdar2015/text_localization/train_icdar2015_label.txt
    ratio_list: [1.0]
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - DetLabelEncode: null
    - CopyPaste: null
    - IaaAugment:
        augmenter_args:
        - type: Fliplr
          args:
            p: 0.5
        - type: Affine
          args:
            rotate:
            - -10
            - 10
        - type: Resize
          args:
            size:
            - 0.5
            - 3
    - EastRandomCropData:
        size:
        - 640
        - 640
        max_tries: 50
        keep_ratio: true
    - MakeBorderMap:
        shrink_ratio: 0.4
        thresh_min: 0.3
        thresh_max: 0.7
        total_epoch: *epoch_num
    - MakeShrinkMap:
        shrink_ratio: 0.4
        min_text_size: 8
        total_epoch: *epoch_num
    - NormalizeImage:
        scale: 1./255.
        mean:
        - 0.485
        - 0.456
        - 0.406
        std:
        - 0.229
        - 0.224
        - 0.225
        order: hwc
    - ToCHWImage: null
    - KeepKeys:
        keep_keys:
        - image
        - threshold_map
        - threshold_mask
        - shrink_map
        - shrink_mask
  loader:
    shuffle: true
    drop_last: false
    batch_size_per_card: 8
    num_workers: 8

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/icdar2015/text_localization/
    label_file_list:
      - ./train_data/icdar2015/text_localization/test_icdar2015_label.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - DetLabelEncode: null
    - DetResizeForTest:
    - NormalizeImage:
        scale: 1./255.
        mean:
        - 0.485
        - 0.456
        - 0.406
        std:
        - 0.229
        - 0.224
        - 0.225
        order: hwc
    - ToCHWImage: null
    - KeepKeys:
        keep_keys:
        - image
        - shape
        - polys
        - ignore_tags
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 1
    num_workers: 2
profiler_options: null

**Comment by liuhongen1234567** [comment]
Hello, PP-OCRv5_mobile_det.yml includes detection images of various sizes in the training set. Generally, you can use the default parameter configuration. If adjustments are needed, you can modify the dimensions of the cropped images. The default is 640*640.
- EastRandomCropData:
size:
- 640
- 640
